src/libsystemd/sd-resolve/test-resolve.c

```diff
@@ -102,9 +104,18 @@ int main(int argc, char *argv[]) {
 
         /* Wait until all queries are completed */
         for (;;) {
-                r = sd_resolve_wait(resolve, (uint64_t) -1);
+                r = sd_resolve_wait(resolve, TEST_TIMEOUT_USEC);
                 if (r == 0)
                         break;
+                if (r == -ETIMEDOUT) {
+                        /* Let's catch time-outs here, so that we can run safely in a CI that has no reliable DNS. Note
+                         * that we invoke exit() directly here, as the stuck NSS call will not allow us to exit
+                         * cleanly. */
+
+                        log_notice_errno(r, "sd_resolve_wait() timed out, but that's OK");
+                        exit(EXIT_SUCCESS);
```

Comment:
EXIT_TEST_SKIP?

Reply:
Answering myself: it's just accounting, so the difference isn't big. But looking at the test, and we _are_ testing a different code path with the timeout, so that's good. After all the tests found a failure mode.